test/TestCases/Expressions/SquareCube.cpp

```diff
@@ -22,14 +22,41 @@
 
 #include <vctr_test_utils/vctr_test_common.h>
 
+template <vctr::is::floatNumber T>
+T sqrt (T x) { return std::sqrt (x); }
+
+template <vctr::is::complexFloatNumber T>
+T sqrt (T x) { return std::sqrt (x); }
+
 template <class T>
 T square (T x) { return x * x; }
 
+template <class T>
+T cube (T x) { return x * x * x; }
+
+TEMPLATE_PRODUCT_TEST_CASE ("Square Root", "[sqrt]", (PlatformVectorOps, VCTR_NATIVE_SIMD), (float, double, std::complex<float>, std::complex<double>) )
+{
+    VCTR_TEST_DEFINES_IN_RANGE (0, 10000, 10)
+
+    const vctr::Vector s = filter << vctr::sqrt << srcA;
+
+    REQUIRE_THAT (s, vctr::EqualsTransformedBy<sqrt> (srcA).withEpsilon (0.000001));
```

Comment:
Should the detection of an `sqrt (negativeFloat)` attempt be also tested?

Reply:
Since that would only throw an assertion, this would not be that meaningful imo. In general, I could think of a pattern where we inject an exception for the VCTR_ASSERT macro and test if that has been hit in unit tests. Let's add that to the backlog